gammapy/maps/geom.py

```diff
@@ -355,19 +355,18 @@ def __init__(self, nodes, interp="lin", name="", node_type="edges", unit=""):
         self._nbin = nbin
 
     def __eq__(self, other):
-        if isinstance(other, self.__class__):
-            return (
-                np.allclose(self._nodes, other._nodes)
+        """Test axis equality. Absolute and relative tolerances of 1e-6 are used"""
+        if not isinstance(other, self.__class__):
+            return NotImplemented
+
+        # TODO: implement an allclose method for MapAxis and call it here
+        other_quantity = other.edges * other.unit
+        other_values = other_quantity.to(self.unit).value
+        return (
+                np.allclose(self.edges, other_values, atol=1e-6, rtol=1e-6)
```

Comment:
This is a can of worms, but yes, I guess giving both an atol and an rtol is the only "safe" thing to do, if you want equality up to floating point noise. So OK to keep as is.